.travis/build-docker.sh

```diff
@@ -17,26 +11,60 @@ GITHASH=`git log -1 --pretty=format:"%h"`
 BUILD_VERSION="${TIMESTAMP}_${GITHASH}${TRAVIS_BRANCH:+:$TRAVIS_BRANCH}"
 echo "BUILD_VERSION $BUILD_VERSION"
 
-# parse options
-while getopts bp option ; do
-  case $option in
-    b) # Pass gridappsd tag to docker-compose
-      # Docker file on travis relative from root.
-      docker build --build-arg TIMESTAMP="${BUILD_VERSION}" -t ${IMAGE}:${TIMESTAMP}_${GITHASH} .
-      ;;
-    p) # Pass gridappsd tag to docker-compose
-      if [ -n "$TAG" -a -n "$ORG" ]; then
-        echo "docker push ${IMAGE}:${TIMESTAMP}_${GITHASH}"
-        docker push ${IMAGE}:${TIMESTAMP}_${GITHASH}
-        docker tag ${IMAGE}:${TIMESTAMP}_${GITHASH} ${IMAGE}:$TAG
-        echo "docker push ${IMAGE}:$TAG"
-        docker push ${IMAGE}:$TAG
-      fi
-      ;;
-    *) # Print Usage
-      usage
-      ;;
-  esac
-done
-shift `expr $OPTIND - 1`
+# Pass gridappsd tag to docker-compose
+docker build --build-arg TIMESTAMP="${BUILD_VERSION}" -t ${IMAGE}:${TIMESTAMP}_${GITHASH} .
+status=$?
+if [ $status -ne 0 ]; then
+  echo "Error: status $status"
+  exit 1
+fi
+
+# To have `DOCKER_USERNAME` and `DOCKER_PASSWORD`
+# filled you need to either use `travis`' cli
+# (https://github.com/travis-ci/travis.rb)
+# and then `travis set ..` or go to the travis
+# page of your repository and then change the
+# environment in the settings pannel.
+
+if [ -n "$DOCKER_USERNAME" -a -n "$DOCKER_PASSWORD" ]; then
+
+  echo " "
+  echo "Connecting to docker"
+
+  echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin
+  status=$?
+  if [ $status -ne 0 ]; then
+    echo "Error: status $status"
+    exit 1
+  fi
+
+  if [ -n "$TAG" -a -n "$ORG" ]; then
+    # Get the built container name, for builds that cross the hour boundary
```

Comment:
The previous logic used the TIMESTAMP to determine the container name for the docker push.  If the build crossed the top of the our the TIMESTAMP would be incorrect and the container would fail to push to docker hub.  With the new method, we lookup the built container name for the docker push.

Reply:
Yes...the thing is it doesn't matter now if it crosses the boundary either way it will look it up correct?

Reply:
Now that the build and push are in one step it is really no longer an issue since the TIMESTAMP is only defined once.  The 
CONTAINER=`docker images --format "{{.Repository}}:{{.Tag}}" ${IMAGE}`
line will make sure it tries to push the container that was created.